library/cpp/grid_creator/binarization.cpp

```diff
@@ -86,12 +86,17 @@ namespace NSplitSelection {
                 throw (yexception() << "Unexpected Nan value.");
             }
         }
-
-        auto firstNanPos = std::remove_if(features.Values.begin(), features.Values.end(), IsNan);
-        if (firstNanPos != features.Values.end()) {
-            if (featureValuesMayContainNans) {
+        
+        if(featureValuesMayContainNans){
+            // Remove all nan to calculate borders in MultiRMSEWithMissingValues
+            auto firstNanPos = std::remove_if(features.Values.begin(), features.Values.end(), IsNan);
+            while(firstNanPos != features.Values.end()){
```

Comment:
What the purpose of `while` here? Sorry, I can't understand why this block of code has been changed.

Reply:
Ok. This while is necessary because all the "nan" value on a target needs to be "removed" before the border calculation using the BestSplit method. The old version was checking for only one nan to remove. 

Whit this value, if is a multiregression with missing value then there is a while loop that check and remove all nan before calculating the borders. Else will stop. Is this answering to your question?
Many thanks 👍🏻

Reply:
The "old" version removes all NaNs. `remove_if` moves all NaNs at the end of vector and returns the first position of NaN (https://en.cppreference.com/w/cpp/algorithm/remove). `erase` deletes this obtained tail of NaNs (https://en.cppreference.com/w/cpp/container/vector/erase).

Reply:
You are right and I see the point. Tested and there is no need of "while" here. Code reverted back. Thank you.